File: questions/Python/Basic/Es2.py
# -- "Easy", weight: '5'
# 'Realizza una funzione "run" '
# 'che moltiplichi due numeri.'

import unittest
from RunningFile import run


class Test(unittest.TestCase):
    def test_1(self):
        self.assertEqual(run(1, 2), 2, 'Errore')

    def test_2(self):
        self.assertEqual(run(0, 2), 0, 'Errore')

    def test_3(self):
        self.assertEqual(run(-1, 2), -2, 'Errore')


unittest.main()

File: questions/Java/Basic/Es2.java
/* -- "Difficult", weight: '1'
/* 'Realizza una classe Esercizio a cui si passi nel costruttore due numeri interi, ' */
/* 'tale classe deve avere una funzione run che ritorni il prodotto dei due numeri' */


public class Es2{
    public static void main(String[] args) {
        Esercizio e1 = new Esercizio(1,2);
        Esercizio e2 = new Esercizio(2,-2);
        if (e1.run() == 2){
            if (e2.run() == -4){
                System.out.println("..\nTest eseguiti con successo.");
            }
            else{
                System.out.println(".F\nUn test fallito.");
            }
        }
        else {
            System.out.println("FF\nTest falliti.");
        }
    }
}